### PR TITLE
Chore/remove resizing

### LINKED ISSUE #39: Configure proto for SchemeGen for better understanding of the TFRecord schema
Demonstrated here: https://github.com/deep-diver/mlops-hf-tf-vision-models/blob/main/notebooks/intermediate.ipynb

### LINKED ISSUE #40: fix: remove image resizing in preprocessing step
since the size of `PETS` dataset is `128x128`, we don't need image resizing in `Transform` component anymore.

### PR DESCRIPTION
**Pull request by deep-diver**
Closes #39, Closes #40. 

enhancing preprocessing step with `schema.pbtxt` by parsing each features in TFRecords into appropriate shapes.

### CODE REVIEW

**Comment by deep-diver** [comment]
@sayakpaul 

sorry, there is a bug to be fixed. will let you know when it is fixed

**Comment by deep-diver** [comment]
@sayakpaul 

This PR is now ready to be reviewed. There are two major changes.
- `Transform` does not use features of `image_shape` and `label_shape`. Instead, `SchemaGenImporter` imports and leverages the predefined schema(`pbtxt`) to determine the shape of `image` and `label` features.

- Since the size of image is already `128x128`, and the UNet model expects the size of input to be `128x128`, the image resizing procedure is removed.

File: training_pipeline/pipeline/configs.py [hunk 1]
```diff
@@ -25,6 +25,7 @@
 PIPELINE_ROOT = os.path.join(OUTPUT_DIR, "tfx_pipeline_output", PIPELINE_NAME)
 
 DATA_PATH = "gs://pets-tfrecords/pets-tfrecords/"
+SCHEMA_PATH = "pipeline/schema.pbtxt"
```
Comment (sayakpaul) [inline training_pipeline/pipeline/configs.py:28]: Is it also possible to supply a GCS path here?

File: training_pipeline/pipeline/local_pipeline.py [hunk 2]
```diff
@@ -52,7 +52,7 @@ def create_pipeline(
     statistics_gen = StatisticsGen(examples=example_gen.outputs["examples"])
     components.append(statistics_gen)
 
-    schema_gen = SchemaGen(statistics=statistics_gen.outputs["statistics"])
+    schema_gen = tfx.components.ImportSchemaGen(schema_file=schema_path)
```
Comment (sayakpaul) [inline training_pipeline/pipeline/local_pipeline.py:55]: Super neat!

File: training_pipeline/pipeline/pipeline.py [hunk 3]
```diff
@@ -62,7 +62,7 @@ def create_pipeline(
     statistics_gen = StatisticsGen(examples=example_gen.outputs["examples"])
     components.append(statistics_gen)
 
-    schema_gen = SchemaGen(statistics=statistics_gen.outputs["statistics"])
+    schema_gen = tfx.components.ImportSchemaGen(schema_file=schema_path)
```
Comment (sayakpaul) [inline training_pipeline/pipeline/pipeline.py:65]: Should we import `ImportSchemaGen` at the top for consistency in style?

File: training_pipeline/pipeline/schema.pbtxt [hunk 4]
```diff
@@ -0,0 +1,64 @@
+feature {
```
Comment (sayakpaul) [inline training_pipeline/pipeline/schema.pbtxt:1]: Maybe add a comment redirecting the readers to guide that shows you how to craft this file?

**Review by sayakpaul — APPROVED** [review]
Extremely neat and useful PR.

File: training_pipeline/pipeline/pipeline.py (hunk 3 above)
Comment (deep-diver) [reply training_pipeline/pipeline/pipeline.py:65]: That is what I thought initially, but it throws an error not to be able to imports with `from tfx.components import ImportSchemaGen`.

Comment (deep-diver) [reply training_pipeline/pipeline/pipeline.py:65]: That is what I thought initially, but it throws an error not to be able to imports with `from tfx.components import ImportSchemaGen`.

File: training_pipeline/pipeline/configs.py (hunk 1 above)
Comment (deep-diver) [reply training_pipeline/pipeline/configs.py:28]: not sure! maybe possible. need to check. 
will let you know soon the result

File: training_pipeline/pipeline/schema.pbtxt (hunk 4 above)
Comment (deep-diver) [reply training_pipeline/pipeline/schema.pbtxt:1]: sure thing

File: training_pipeline/pipeline/configs.py (hunk 1 above)
Comment (sayakpaul) [reply training_pipeline/pipeline/configs.py:28]: Did you get a chance to see this?

Comment (deep-diver) [reply training_pipeline/pipeline/configs.py:28]: just checked right now, and it works with GCS as well :)

**Pull request merged**